lua/nvim-treesitter/query.lua

```diff
@@ -13,12 +13,8 @@ local query_cache = caching.create_buffer_cache()
 M.base_language_map = {
   cpp = {'c'},
   typescript = {'javascript'},
-  tsx = {'typescript', 'javascript'},
-}
-
-M.query_extensions = {
-  javascript = { 'jsx' },
-  tsx = {'jsx'}
+  jsx = {'javascript'},
```

Comment:
i believe this should be `javascript = { 'jsx' }`

Reply:
Should we do the same for typescript? Or do we need both relationshipts? `tsx = {'typescript'}` and `typescript = {'tsx'}`

Reply:
tsx will be loaded when encountering typescriptreact, juste like javascriptreact will load javascript which will load the jsx query. Typescript parser does understand jsx queries so we should not do typescript = { 'tsx' }

Reply:
Changed!